tests/models/bart/test_modeling_tf_bart.py

```diff
@@ -138,6 +150,23 @@ def check_decoder_model_past_large_inputs(self, config, inputs_dict):
         # test that outputs are equal for slice
         tf.debugging.assert_near(output_from_past_slice, output_from_no_past_slice, rtol=1e-3)
 
+    def create_and_check_bart_xla_generate_fast(self, config, input_ids, *args):
+        config.eos_token_id = None
+        config.max_length = 10
+        config.do_sample = False
+        config.num_beams = 1
+        model = TFBartForConditionalGeneration(config=config)
+
+        # make sure there are no pad tokens in prompt
+        input_ids = tf.where(input_ids != config.pad_token_id, input_ids, config.pad_token_id - 1)
```

Comment:
If I understand correctly, we want to avoid the pad tokens in the middle of the prompt(s).

The (normal) padded prompt(s) would still be fine, right?

Reply:
Yes -- this is because there are some if/elses in the generate input preparation that trigger if there are pad tokens, but they expect them to be in the beginning or end of the inputs (ideally, there should be a corresponding attention mask). In this test, the input is randomly generated, so they can be in the middle, so we either remove it or create the corresponding mask.